src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -94,6 +94,14 @@ internal partial class Binder
 
                 foreach (var p in parameterSyntaxList.Value)
                 {
+                    if (p.AttributeLists.Count > 0 && syntax.Kind() == SyntaxKind.AnonymousMethodExpression)
```

Comment:
One can always build the tree manually with factory. Given your findings, I see no reasons for the check to be conditional.